org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/validation/XbaseValidator.java

```diff
@@ -1355,6 +1355,24 @@ protected boolean isLocallyUsed(EObject target, EObject containerToFindUsage) {
 			return true;
 		}
 		Collection<Setting> usages = XbaseUsageCrossReferencer.find(target, containerToFindUsage);
+		// field are used when they are not used as the left operand of an assignment operator.
+		if (target instanceof JvmField) {
+			for (final Setting usage : usages) {
+				EObject object = usage.getEObject();
+				while (object instanceof XMemberFeatureCall) {
```

Comment:
The problem is that `usage.getEObject()` does not gives an `XAssignment` object, but a `XMemberFeatureCall` object, in the case of `a.b = c`.
Without this loop, the following test on `XAssignement` fails.
Calling `EcoreUtil2.getContainerOfType` for retrieving the XAssignment is not a solution because I would like to detect a member feature call as left operand of assignment. Any other pattern should be ignored.

Reply:
I'm not convinced yet. I just had a look at the parsed AST and the field is the feature of an XAssignment, not of a XMemberFeatureCall (for the relevant cases). Maybe I'm missing smth. Did you add a unit test for that?

Reply:
The basic assumption is: a direct usage (or through a sequence of XMemberFeatureCall) of a field as feature of an XAssignement means that the field is not read but write.

Yes, I have unit tests but with Xtend and my own DSL. But, I do not known how to test JvmField-related things within Xtext extras tools only. Do you have an example of a unit test within xtext-extras that I could adapt?

Reply:
It would be great if you could add a bunch of test cases for Xtend in the xtend repo.